proposals/src/header.rs

```diff
@@ -163,6 +175,24 @@ impl From<[u8; Self::LENGTH]> for TypedChainId {
     }
 }
 
+impl From<[u8; Self::LENGTH + 2]> for TypedChainId {
+    fn from(bytes: [u8; Self::LENGTH + 2]) -> Self {
+        let ty = [bytes[2], bytes[3]];
+        let ty = u16::from_be_bytes(ty);
+        let id = u32::from_be_bytes([bytes[4], bytes[5], bytes[6], bytes[7]]);
+        match ty {
+            0x0100 => TypedChainId::Evm(id),
+            0x0200 => TypedChainId::Substrate(id),
+            0x0301 => TypedChainId::PolkadotParachain(id),
+            0x0302 => TypedChainId::KusamaParachain(id),
+            0x0303 => TypedChainId::RococoParachain(id),
+            0x0400 => TypedChainId::Cosmos(id),
+            0x0500 => TypedChainId::Solana(id),
+            _ => Self::None,
+        }
+    }
+}
```

Comment:
Could we also have the `impl From<u64> for TypedChainId`? it should be fairly easy as converting the `u64` to bytes and calling the `TypedChainid::from(u64_bytes)`

Reply:
Added!